<?xml version="1.0" encoding="UTF-8" ?>
<!-- SPDX-License-Identifier: GPL-3.0-or-later -->
<ui version="4.0">
  <class>DlgBPMDetect</class>
  <widget class="QWidget" name="DlgBPMDetect">
    <property name="geometry">
      <rect>
        <x>0</x>
        <y>0</y>
        <width>700</width>
        <height>400</height>
      </rect>
    </property>
    <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
      </sizepolicy>
    </property>
    <property name="minimumSize">
      <size>
        <width>700</width>
        <height>400</height>
      </size>
    </property>
    <property name="windowTitle">
      <string>BPM Detect</string>
    </property>
    <layout class="QGridLayout">
      <property name="margin">
        <number>5</number>
      </property>
      <item row="0" column="0">
        <layout class="QHBoxLayout">
          <item>
            <widget class="QPushButton" name="btnAddFiles">
              <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="minimumSize">
                <size>
                  <width>100</width>
                  <height>0</height>
                </size>
              </property>
              <property name="text">
                <string>Add &amp;files</string>
              </property>
              <property name="shortcut">
                <string>Alt+F</string>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QPushButton" name="btnAddDir">
              <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="minimumSize">
                <size>
                  <width>100</width>
                  <height>0</height>
                </size>
              </property>
              <property name="text">
                <string>Add director&amp;y</string>
              </property>
              <property name="shortcut">
                <string>Alt+Y</string>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QPushButton" name="btnRemoveSelected">
              <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="minimumSize">
                <size>
                  <width>100</width>
                  <height>0</height>
                </size>
              </property>
              <property name="toolTip">
                <string>Remove selected tracks</string>
              </property>
              <property name="text">
                <string>&amp;Remove</string>
              </property>
              <property name="shortcut">
                <string>Alt+R</string>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QPushButton" name="btnClearList">
              <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="minimumSize">
                <size>
                  <width>100</width>
                  <height>0</height>
                </size>
              </property>
              <property name="text">
                <string>Cl&amp;ear list</string>
              </property>
              <property name="shortcut">
                <string>Alt+E</string>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QLabel" name="lblFormat">
              <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="text">
                <string>BPM tag format:</string>
              </property>
              <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
              <property name="wordWrap">
                <bool>false</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QComboBox" name="cbFormat">
              <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="minimumSize">
                <size>
                  <width>100</width>
                  <height>0</height>
                </size>
              </property>
              <property name="maximumSize">
                <size>
                  <width>100</width>
                  <height>32767</height>
                </size>
              </property>
              <item>
                <property name="text">
                  <string>0.00</string>
                </property>
              </item>
              <item>
                <property name="text">
                  <string>0.0</string>
                </property>
              </item>
              <item>
                <property name="text">
                  <string>0</string>
                </property>
              </item>
              <item>
                <property name="text">
                  <string>000.00</string>
                </property>
              </item>
              <item>
                <property name="text">
                  <string>000.0</string>
                </property>
              </item>
              <item>
                <property name="text">
                  <string>000</string>
                </property>
              </item>
              <item>
                <property name="text">
                  <string>00000</string>
                </property>
              </item>
            </widget>
          </item>
        </layout>
      </item>
      <item row="1" column="0">
        <widget class="QDropListView" name="TrackList">
          <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
          </property>
          <property name="acceptDrops">
            <bool>true</bool>
          </property>
          <property name="selectionMode">
            <enum>QAbstractItemView::ExtendedSelection</enum>
          </property>
          <property name="rootIsDecorated">
            <bool>false</bool>
          </property>
          <property name="itemsExpandable">
            <bool>false</bool>
          </property>
          <property name="sortingEnabled">
            <bool>true</bool>
          </property>
          <property name="allColumnsShowFocus">
            <bool>true</bool>
          </property>
          <property name="columnCount">
            <number>0</number>
          </property>
        </widget>
      </item>
      <item row="2" column="0">
        <widget class="QGroupBox" name="grpProgress">
          <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
            </sizepolicy>
          </property>
          <property name="title">
            <string>Progress</string>
          </property>
          <layout class="QVBoxLayout">
            <property name="spacing">
              <number>2</number>
            </property>
            <property name="margin">
              <number>2</number>
            </property>
            <item>
              <widget class="QLabel" name="lblCurrentTrack">
                <property name="sizePolicy">
                  <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                  </sizepolicy>
                </property>
                <property name="maximumSize">
                  <size>
                    <width>570</width>
                    <height>32767</height>
                  </size>
                </property>
                <property name="toolTip">
                  <string>Current track</string>
                </property>
                <property name="text">
                  <string />
                </property>
                <property name="wordWrap">
                  <bool>false</bool>
                </property>
              </widget>
            </item>
            <item>
              <widget class="QProgressBar" name="TotalProgress">
                <property name="value">
                  <number>0</number>
                </property>
              </widget>
            </item>
          </layout>
        </widget>
      </item>
      <item row="3" column="0">
        <layout class="QHBoxLayout">
          <item>
            <widget class="QPushButton" name="btnAbout">
              <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="minimumSize">
                <size>
                  <width>90</width>
                  <height>0</height>
                </size>
              </property>
              <property name="text">
                <string>&amp;About</string>
              </property>
              <property name="shortcut">
                <string>Alt+A</string>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QLabel" name="lblMin">
              <property name="text">
                <string>Min:</string>
              </property>
              <property name="wordWrap">
                <bool>false</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QSpinBox" name="spMin">
              <property name="toolTip">
                <string>Minimum BPM</string>
              </property>
              <property name="minimum">
                <number>50</number>
              </property>
              <property name="maximum">
                <number>120</number>
              </property>
              <property name="value">
                <number>80</number>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QLabel" name="lblMax">
              <property name="text">
                <string>Max:</string>
              </property>
              <property name="wordWrap">
                <bool>false</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QSpinBox" name="spMax">
              <property name="toolTip">
                <string>Maximum BPM</string>
              </property>
              <property name="minimum">
                <number>130</number>
              </property>
              <property name="maximum">
                <number>220</number>
              </property>
              <property name="value">
                <number>185</number>
              </property>
            </widget>
          </item>
          <item>
            <spacer>
              <property name="orientation">
                <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType">
                <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0">
                <size>
                  <width>20</width>
                  <height>21</height>
                </size>
              </property>
            </spacer>
          </item>
          <item>
            <widget class="QCheckBox" name="chbSkipScanned">
              <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="minimumSize">
                <size>
                  <width>150</width>
                  <height>0</height>
                </size>
              </property>
              <property name="toolTip">
                <string>Do not redetect BPM</string>
              </property>
              <property name="text">
                <string>S&amp;kip scanned tracks</string>
              </property>
              <property name="shortcut">
                <string>Alt+K</string>
              </property>
              <property name="checked">
                <bool>true</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QCheckBox" name="chbSave">
              <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="minimumSize">
                <size>
                  <width>75</width>
                  <height>0</height>
                </size>
              </property>
              <property name="toolTip">
                <string>Save BPM to tag</string>
              </property>
              <property name="text">
                <string>Sa&amp;ve</string>
              </property>
              <property name="shortcut">
                <string>Alt+V</string>
              </property>
              <property name="checked">
                <bool>true</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QPushButton" name="btnStart">
              <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="minimumSize">
                <size>
                  <width>90</width>
                  <height>0</height>
                </size>
              </property>
              <property name="text">
                <string>St&amp;art</string>
              </property>
              <property name="shortcut">
                <string>Alt+A</string>
              </property>
              <property name="autoDefault">
                <bool>true</bool>
              </property>
              <property name="default">
                <bool>true</bool>
              </property>
            </widget>
          </item>
        </layout>
      </item>
    </layout>
  </widget>
  <layoutdefault spacing="6" margin="11" />
  <customwidgets>
    <customwidget>
      <class>QDropListView</class>
      <extends>QTreeWidget</extends>
      <header>widgets/qdroplistview.h</header>
    </customwidget>
  </customwidgets>
  <includes>
    <include location="local">widgets/qdroplistview.h</include>
  </includes>
  <resources>
    <include location="resource.qrc" />
  </resources>
  <connections>
    <connection>
      <sender>btnAddFiles</sender>
      <signal>clicked()</signal>
      <receiver>DlgBPMDetect</receiver>
      <slot>slotAddFiles()</slot>
      <hints>
        <hint type="sourcelabel">
          <x>20</x>
          <y>20</y>
        </hint>
        <hint type="destinationlabel">
          <x>20</x>
          <y>20</y>
        </hint>
      </hints>
    </connection>
    <connection>
      <sender>btnAddDir</sender>
      <signal>clicked()</signal>
      <receiver>DlgBPMDetect</receiver>
      <slot>slotAddDir()</slot>
      <hints>
        <hint type="sourcelabel">
          <x>20</x>
          <y>20</y>
        </hint>
        <hint type="destinationlabel">
          <x>20</x>
          <y>20</y>
        </hint>
      </hints>
    </connection>
    <connection>
      <sender>btnRemoveSelected</sender>
      <signal>clicked()</signal>
      <receiver>DlgBPMDetect</receiver>
      <slot>slotRemoveSelected()</slot>
      <hints>
        <hint type="sourcelabel">
          <x>20</x>
          <y>20</y>
        </hint>
        <hint type="destinationlabel">
          <x>20</x>
          <y>20</y>
        </hint>
      </hints>
    </connection>
    <connection>
      <sender>btnAbout</sender>
      <signal>clicked()</signal>
      <receiver>DlgBPMDetect</receiver>
      <slot>slotShowAbout()</slot>
      <hints>
        <hint type="sourcelabel">
          <x>20</x>
          <y>20</y>
        </hint>
        <hint type="destinationlabel">
          <x>20</x>
          <y>20</y>
        </hint>
      </hints>
    </connection>
    <connection>
      <sender>btnClearList</sender>
      <signal>clicked()</signal>
      <receiver>DlgBPMDetect</receiver>
      <slot>slotClearTrackList()</slot>
      <hints>
        <hint type="sourcelabel">
          <x>20</x>
          <y>20</y>
        </hint>
        <hint type="destinationlabel">
          <x>20</x>
          <y>20</y>
        </hint>
      </hints>
    </connection>
  </connections>
</ui>
